### PR TITLE
feat: Improvements

### PR DESCRIPTION
**Pull request by prat-gpt**
Issues addressed: https://github.com/solidity-labs-io/forge-proposal-simulator/issues/81, https://github.com/solidity-labs-io/forge-proposal-simulator/issues/79, https://github.com/solidity-labs-io/forge-proposal-simulator/issues/77, https://github.com/solidity-labs-io/forge-proposal-simulator/issues/76 and https://github.com/solidity-labs-io/forge-proposal-simulator/issues/66

### CODE REVIEW

**Comment by ElliotFriedman** [comment]
This PR should enforce a line length of 80 characters to keep things clean.

Can we remove the `package.json`, `package-lock.json` and other node related files from the repo entirely?

File: src/proposals/GovernorBravoProposal.sol [hunk 1]
```diff
@@ -48,21 +38,12 @@ abstract contract GovernorBravoProposal is Proposal {
 
     /// @notice Check if there are any on-chain proposals that match the
     /// proposal calldata
-    function checkOnChainCalldata()
-        public
-        view
-        override
-        returns (bool calldataExist)
-    {
```
Comment (ElliotFriedman) [inline src/proposals/GovernorBravoProposal.sol:56]: why remove this function?

**Comment by prat-gpt** [comment]
> This PR should enforce a line length of 80 characters to keep things clean.

I will make this update
> Can we remove the `package.json`, `package-lock.json` and other node related files from the repo entirely?

We use solhint and husky so I don't think we can remove these files

**Comment by ElliotFriedman** [comment]
> > Can we remove the `package.json`, `package-lock.json` and other node related files from the repo entirely?
> 
> We use solhint and husky so I don't think we can remove these files

Can we get rid of solhint and husky and solely rely on foundry linting?

We can just roll our own precommit shell script that runs `forge fmt` on all files, and re-adds the files staged for commit

File: docs/guides/customizing-proposal.md [hunk 2]
```diff
@@ -534,90 +534,90 @@ Copy all address arbitrum address from [Addresses.json](https://github.com/solid
...
-    },
-    {
-        "addr": "0x2c9c0F10E3F8820544522df210dFb0A2BbC75147",
-        "name": "DEPLOYER_EOA",
-        "chainId": 1,
-        "isContract": false
-    }
+  {
+    "addr": "0xE6841D92B0C345144506576eC13ECf5103aC7f49",
+    "chainId": 1,
```
Comment (prat-gpt) [inline docs/guides/customizing-proposal.md]: remove chainid

File: mocks/MockAuction.sol [hunk 3]
```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.8.0;
+
+contract MockAuction {
```
Comment (prat-gpt) [inline mocks/MockAuction.sol:4]: add comment to specify not to use in production in all mocks

File: docs/guides/customizing-proposal.md
Comment (prat-gpt) [inline docs/guides/customizing-proposal.md:1]: Add docs for splitting address json

File: test/Addresses.t.sol [hunk 4]
```diff
@@ -303,10 +321,10 @@ contract TestAddresses is Test {
...
         // update addresses object with updated addresses.json
-        addresses = new Addresses(addressesPath);
+        addresses = new Addresses(addressesFolderPath, chainIds);
 
         // check TEST1 address is updated
-        assertEq(addresses.getAddress("TEST1", 123), test1);
+        assertEq(addresses.getAddress("TEST1", 11155111), test1);
 
         // check TEST2 address is added
         assertEq(addresses.getAddress("TEST2"), test2);
```
Comment (prat-gpt) [inline test/Addresses.t.sol:334]: add a assert for third network

Comment (prat-gpt) [reply test/Addresses.t.sol:334]: expectRevert to try to get the address not on current chain ID

File: docs/overview/architecture/addresses.md
Comment (prat-gpt) [inline docs/overview/architecture/addresses.md:1]: Add a section showing example folder tree structure

File: docs/guides/introduction.md [hunk 5]
```diff
@@ -22,12 +22,12 @@ echo @forge-proposal-simulator=lib/forge-proposal-simulator/ >> remappings.txt
 
 ### Step 3: Addresses File
 
-Create a JSON file following the standard on [Addresses](../overview/architecture/addresses.md). It is recommended to keep the addresses file in a separate folder, for example, `./addresses/Addresses.json`. Also allow read access for this file in `foundry.toml`.
+Create a JSON file following the standard on [Addresses](../overview/architecture/addresses.md). It is recommended to keep the JSON file in a separate folder, for example, `./addresses/31337.json`. Also allow read access for this file in `foundry.toml`.
```
Comment (prat-gpt) [inline docs/guides/introduction.md]: add that the file name should be same as chain id

File: docs/guides/introduction.md [hunk 6]
```diff
@@ -22,12 +22,12 @@ echo @forge-proposal-simulator=lib/forge-proposal-simulator/ >> remappings.txt
 
 ### Step 3: Addresses File
 
-Create a JSON file following the standard on [Addresses](../overview/architecture/addresses.md). It is recommended to keep the addresses file in a separate folder, for example, `./addresses/Addresses.json`. Also allow read access for this file in `foundry.toml`.
+Create a JSON file following the standard on [Addresses](../overview/architecture/addresses.md). It is recommended to keep the JSON file in a separate folder, for example, `./addresses/31337.json`. The name of the JSON file would be the same as the network id. If there are multiple networks, addresses should be added in the JSON files corresponding to their network. Also, allow read access for this file in `foundry.toml`.
```
Comment (ElliotFriedman) [inline docs/guides/introduction.md]:
```suggestion
Create a JSON file following the standard on [Addresses](../overview/architecture/addresses.md). It is recommended to keep the JSON file in a separate folder, for example, `./addresses/31337.json`. The name of the JSON file should be the same as the network id. If there are multiple networks, addresses should be added in the JSON files corresponding to their network. Be sure to allow read access for this file in `foundry.toml`.
```

**Review by ElliotFriedman — APPROVED** [review]
LGTM!

**Pull request merged**